packages/core/core/src/services/document-service/transform/relations/extract/data-ids.ts

```diff
@@ -82,7 +83,14 @@ const extractDataIds = (
         // TODO: Handle morph relations (they have multiple targets)
         if (!target) return;
 
-        extractedIds.forEach((id) => idMap.add(target, id as string, opts.locale));
+        extractedIds.forEach((id) =>
+          idMap.add({
+            uid: target,
+            documentId: id as string,
```

Comment:
We should avoid 'as', here `ID` is not just string. Can we make the idMap support `ID`, or else cast it as a string? (might not be relevant depending on the answer to the following question re: object ids for db id)

Reply:
makes sense, I will keep it as ID for now :) I have had this issue many times before, and probably now its better to use the same ID everywhere